Qt fixes + protobuf fixes

m-simonelli opened the pull request:
Replaced some deprecated functions with newer equivalents. Updated protobuf files to specify proto2 syntax. Updated contact id validator tests from v2 onion -> v3 onion.

m-simonelli committed:
fixed UI for V3 onions, fixed tests for V3 onions

m-simonelli committed:
libtego-ui: specify protobuf syntax version

m-simonelli committed:
libtego_ui: updated deprecated qrand to QRandomGenerator::global()->generate64()
tego_ui: updated deprecated qsrand to QRandomGenerator::global()->seed()

morganava commented on src/libtego_ui/protocol/Connection.cpp:
```diff
@@ -465,7 +467,7 @@ int ConnectionPrivate::availableOutboundChannelId()
     // Find an unused id, trying a maximum of 100 times, using a random step to avoid collision
     for (int i = 0; i < 100 && channels.contains(nextOutboundChannelId); i++) {
-        nextOutboundChannelId += 1 + (qrand() % 200);
+        nextOutboundChannelId += 1 + ((int)QRandomGenerator::global()->generate64() % 200);
```
If we're looking for an int on [0,199] than we should use QRandomGenerator::bounded(0, 199) ( see: https://doc.qt.io/qt-5/qrandomgenerator.html#bounded-4 )

m-simonelli replied:
QRandomGenerator::global()->bounded(200) works as well. `highest` argument to bounded is exclusive, so this will generate from 0-199 inclusive.

m-simonelli committed:
libtego_ui: change qrandomgenerator from generate64 to bounded

morganava commented on src/libtego_ui/protocol/Connection.cpp:
```diff
@@ -34,6 +34,8 @@
 #include "ControlChannel.h"
 #include "utils/Useful.h"
 
+#include <QRandomGenerator>
```
put dependency (ie Qt, stdlib, etc) headers  in the precompiled header

morganava commented on src/tego_ui/main.cpp:
```diff
@@ -38,6 +38,7 @@
 #include "utils/CryptoKey.h"
 #include "utils/SecureRNG.h"
 #include "utils/Settings.h"
+#include <QRandomGenerator>
```
place in precompiled header

morganava reviewed (changes requested):
looks good to me apart from my comments

m-simonelli committed:
libtego_ui: moved qt include to precomp

m-simonelli committed:
Merge branch 'qt-fixes' of https://github.com/m-simonelli/tego-core into qt-fixes

m-simonelli committed:
tego_ui: QRandomGenerator::global should not be manually seeded.

QRandomGenerator::global is already seeded by QT, and attempting to seed it manually causes a crash. See https://doc.qt.io/qt-5/qrandomgenerator.html#global
